src/main/java/org/broadinstitute/hellbender/utils/IndexUtils.java

```diff
@@ -0,0 +1,44 @@
+package org.broadinstitute.hellbender.utils;
+
+import htsjdk.tribble.Tribble;
+import htsjdk.tribble.index.Index;
+import htsjdk.tribble.index.IndexFactory;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.hellbender.exceptions.UserException;
+
+import java.io.File;
+
+public final class IndexUtils {
+    private IndexUtils(){}
+
+    private static final Logger logger = LogManager.getLogger(IndexUtils.class);
+
+    /**
+     * Load the index from disk, checking for out of date indexes and old versions
+     * @return an Index, or null if we're unable to load
+     */
+    public static Index loadTribbleIndex(final File featureFile) {
+        Utils.nonNull(featureFile);
+        final File indexFile = Tribble.indexFile(featureFile);
+        if (! indexFile.canRead()) {
+            return null;
+        }
+        logger.debug("Loading Tribble index from disk for file " + featureFile);
+        final Index index = IndexFactory.loadIndex(indexFile.getAbsolutePath());
+        checkIndexModificationTime(featureFile, indexFile, index);
+        return index;
+    }
+
+    /**
+     * @throws UserException the index not up-to date (ie older than the feature file) and is not the current version.
+     */
+    public static void checkIndexModificationTime(final File featureFile, final File indexFile, final Index index) {
+        if (! index.isCurrentVersion()) {
+            // we've loaded an old version of the index, we want to remove it <-- currently not used, but may re-enable
+            throw new UserException("Index file " + indexFile + " is out of date (old version). Use IndexFeatureFile to make an index.");
+        } else if (indexFile.lastModified() < featureFile.lastModified()) {
+            throw new UserException("Index file " + indexFile + " is out of date (index older than input file). Use IndexFeatureFile to make an index.");
```

Comment:
@vdauwera and @eitanbanks have made it clear that by default the GATK shouldn't blow up if an index is older than the file it indexes (see https://github.com/broadinstitute/gatk/issues/1683). For now, can you turn this into a scary-looking `logger.warn()` message? 

When we do https://github.com/broadinstitute/gatk/issues/1683 in a future branch we can provide an argument to do stricter index checking and blow up here.